pesuacademy/pesuacademy.py

```diff
@@ -1,186 +1,176 @@
-        """
-        if not self._authenticated:
-            raise AuthenticationError("You need to authenticate first.")
-        profile_info = self.page_handler.get_profile()
-        return profile_info
-
-    def courses(self, semester: Optional[int] = None) -> dict[int, list[Course]]:
-        """
-        Get the courses of the currently authenticated user.
-
-        :param semester: The semester number. If not provided, all courses across all semesters are returned.
-        :return: The course information for the given semester.
-        """
-        if not self._authenticated:
-            raise AuthenticationError("You need to authenticate first.")
-        courses_info = self.page_handler.get_courses(semester)
-        return courses_info
-
-    def attendance(self, semester: Optional[int] = None) -> dict[int, list[Course]]:
-        """
-        Get the attendance in courses of the currently authenticated user.
-
-        :param semester: The semester number. If not provided, attendance across all semesters are returned.
-        :return: The attendance information for the given semester.
-        """
-        if not self._authenticated:
-            raise AuthenticationError("You need to authenticate first.")
-        attendance_info = self.page_handler.get_attendance(semester)
-        return attendance_info
-
-    def seating_information(self) -> list[SeatingInformation]:
-        """
-        Get the seating information of the currently authenticated user.
-
-        :return: The seating information.
-        """
-        if not self._authenticated:
-            raise AuthenticationError("You need to authenticate first.")
-        seating_info = self.page_handler.get_seating_info()
-        return seating_info
-
-    def announcements(
-        self, start_date: Optional[str] = None, end_date: Optional[str] = None
-    ) -> list[Announcement]:
-        """
-        Get the announcements from the PESU Academy website.
+    def __init__(self, client: PesuAcademyClient):
+        """
+        Initializes the PESUAcademy session. This method is not meant to be called directly.
+        Please use the `PESUAcademy.login()` class method to create an instance.
+
+        Args:
+            client (PesuAcademyClient): An authenticated instance of the core client.
+        """
+        self._client = client
+
+    @classmethod
+    async def login(cls, username: str, password: str) -> "PESUAcademy":
+        """
+        Creates, authenticates, and returns an active PESUAcademy session.
+        This is the primary way to start using the library.
+
+        Args:
+            username (str): The user's SRN, PRN, or other login identifier.
+            password (str): The user's password.
+
+        Returns:
+            PESUAcademy: An authenticated instance ready to make requests.
+        """
+        client = PesuAcademyClient()
+        await client.login(username, password)
+        return cls(client)

```

Comment:
This function should allow logging in without passing arguments by reading from the environments. Credentials are usually read from the env in production systems, not passed as arguments.